executor/executor_linux.h

```diff
@@ -186,9 +197,14 @@ static void cover_reset(cover_t* cov)
 			fail("cover_reset: current_thread == 0");
 		cov = &current_thread->cov;
 	}
-	cover_unprotect(cov);
-	*(uint64*)cov->data = 0;
-	cover_protect(cov);
+	if (flag_read_only_coverage) {
+		if (ioctl(cov->fd, KCOV_RESET_TRACE, 0))
+			exitf("KCOV_RESET_TRACE failed");
```

Comment:
Nit: If we want this error to be reported as a crash, `fail`/`failmsg` would work better here.